packages/workers-utils/src/config/validation.ts

```diff
@@ -2065,6 +2065,22 @@ const validateVars =
 				);
 				isValid = false;
 			} else {
+				// Check each var value for invalid types (e.g., Date objects)
+				// and convert them to strings automatically
+				for (const [varName, varValue] of Object.entries(value)) {
+					if (varValue instanceof Date) {
+						// Convert Date to ISO date string (YYYY-MM-DD format)
+						const dateString = varValue.toISOString().split("T")[0];
```

Comment:
Is this correct to convert to "YYYY-MM-DD" ?

TOML supports [multiple distinct date formats](https://toml.io/en/v1.0.0#offset-date-time) and we might end up with a string that is not what is in the TOML file.

At least we should have more tests and test the round trip transformation from the string in TOML to the string value of the var

Reply:
I'm pretty sure the real implementation creates `smol-toml` `TomlDate` objects which are `Date` subclasses. They *can* round-trip the string values. The problem is that the test is using JS Date objects. Really they should use `parseToml` to get the correct objects.